services/chat/api.go

```diff
@@ -149,7 +149,9 @@ func (api *API) getChannelGroups(ctx context.Context, channelGroupID string) (ma
 			if !chat.IsActivePersonalChat() {
 				continue
 			}
-			totalUnviewedMessageCount += int(chat.UnviewedMessagesCount)
+			if !chat.Muted || chat.UnviewedMentionsCount > 0 {
```

Comment:
Append total unviewed messages when _chat is muted and has mentions_ seems a bit off to me. What is the use case? Why not simply `if !chat.Muted {`?

Reply:
We still show badges when someone is mentioned.

I'm pretty sure currently, it would still work if I only put `if !chat.Muted {`, but as I found out with the chat badges and unread state, it's safer to say we have unviewed messages when there is a mention (which is true, if we have a mention, we have unviewed messages)

Reply:
Shouldn't the logic be as follows then?

```
    if !chat.Muted {
        totalUnviewedMessageCount += int(chat.UnviewedMessagesCount)
    }

    if chat.UnviewedMentionsCount > 0 {
        totalUnviewedMessageCount += int(chat.UnviewedMentionsCount)
        totalUnviewedMentionsCount += int(chat.UnviewedMentionsCount)
    }
```